README.md

```diff
@@ -0,0 +1,142 @@
+# Pactum
+## The HTTP-API specification sketchbook for pythonistas
+
+[![Circle CI](https://circleci.com/gh/olist/pactum.svg?style=svg)](https://circleci.com/gh/olist/pactum)
+
+```shell
+pip install pactum
+```
+
+With `Pactum` you can **specify** HTTP-APIs using pure python.
+
+Pactum is easy to use, easy to extend and easy to contribute:
+
+### Easy to use
+
+The only requirements to start writing an API specification with `pactum`
+is `pactum` package itself and some knowledge of python.
+
+```python
+import pactum
+
+class MyAPI(API):
+    name = 'My API'
+    versions = [...]
+```
+
+### Easy to extend
+
+Using the [visitor pattern](http://wiki.c2.com/?VisitorPattern) you can create
+exporters and extensions for any format or service you want.
+
+Take a look at [pactum/exporters/openapi.py](https://github.com/olist/pactum/blob/master/pactum/exporters/openapi.py).
+
+### Architecture
+
+Always keep [this diagram](https://github.com/olist/pactum/wiki/Architecture-Diagram) in mind when defining your APIs.
+
+
+### Tutorial
+
+Create a file called specs.py and start defining your API.
+
+You can define a `Resource` object for your API.
+
+```python
+from pactum import Action, API, Resource, Response, Version
+from pactum import fields, verbs
+
+class OrderResource(Resource):
+    fields = [
+        fields.IntegerField(name='code', required=True),
+        fields.TimestampField(name='created_at'),
+        fields.StringField(name='item')
+    ]
+resource = OrderResource()
+
+error_resource = Resource(
+    name = 'ErrorResource'
+    fields = [fields.StringField(name='error', required=False)]
+)
+```
+You can define any element of your specification by calling it directly as in
+`error_resource` or by class definition as in `MyResource` and then calling it.
+
+
+List resources are definitions of lists of the same resource.
+```python
+list_order_resource = ListResource(resource=resource)
+```
+
+You can define `Response` objects with `status`, `description`(optional)  a
+`header`(optional) and a `Resource`/`ListResource` object as `body` (optionally)...
+
+```python
+ok_response = Response(
+    status=200, description='Here is your orders list.', body=list_resource
+)
+
+error_response = Response(status=400, resource=error_resource, headers=[('Content-type': 'application-json')])
```

Comment:
does the headers argument also accepts dicts?

Reply:
No. HTTP spec allows header duplication. We could handle this using an external lib like multidict but I prefer to not add external requirements at this moment.

Reply:
There's nothing that says that your `headers` can't be a dictionary. But as explained by @osantana, headers can be duplicated by the HTTP specification. For the openapi exporter, for example, we casted headers to a dict because the specification for openapi says it's an "object".